public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx

```diff
@@ -134,26 +108,38 @@ function RuleEditorSectionBody({
 
       <RoutingOptionDescription manualRouting={manualRouting} />
 
-      {manualRouting ? (
-        <div className={styles.simplifiedRouting}>
-          <SimplifiedRouting />
-        </div>
-      ) : (
-        shouldRenderPreview && (
-          <NotificationPreview
-            alertQueries={queries}
-            customLabels={labels}
-            condition={condition}
-            folder={folder}
-            alertName={alertName}
-            alertUid={alertUid}
-          />
-        )
-      )}
+      {manualRouting ? <SimplifiedRouting /> : <AutomaticRooting shouldRenderPreview={true} alertUid={alertUid} />}
     </Stack>
   );
 }
 
+interface AutomaticRootingProps {
+  shouldRenderPreview: boolean;
+  alertUid?: string;
+}
+
+function AutomaticRooting({ shouldRenderPreview, alertUid }: AutomaticRootingProps) {
+  const { watch } = useFormContext<RuleFormValues>();
+  const [labels, queries, condition, folder, alertName] = watch([
+    'labels',
+    'queries',
+    'condition',
+    'folder',
+    'name',
+    'manualRouting',
+  ]);
+  return shouldRenderPreview ? (
```

Comment:
This seems a bit odd to me to pass a boolean in to this component for conditional rendering, can we hoist the logic for not rendering this component to the parent?

Reply:
💯 done!